Advanced-Recruit-Training/src/LogFileReader.cpp

```diff
@@ -57,9 +58,27 @@ bool LogFileReader::readAll(const QString& fileName)
  * that the conversion from string to double is sucessful.*/
 bool LogFileReader::parseLine(const QString& line, BatteryData& batteryData) const
 {
+    { // check for exact format "hh:mm:ss.zzz, voltage, current"
+        // time must be hh:mm:ss.zzz
+        QString regexHour = "([0-1]\\d|2[0-3])", regexMinSec = "[0-5]\\d", regexMSec = "\\d+";
+        QString regexTime = regexHour + ":" + regexMinSec + ":" + regexMinSec + "\\." + regexMSec;
+
+        // voltage must be a positive number, current can be any number
+        QString regexVoltage = "\\d+.\\d+", regexCurrent = "(-\\d|\\d)+\\.\\d+";
+
+        // putting it all together
+        QString regexItem = QString("(^%1, %2, %3$)").arg(regexTime, regexVoltage, regexCurrent);
+
+        QRegExp rxItem(regexItem);
```

Comment:
Compiling a regex is actually a very expensive task, recompiling it every time has a fairly large hit on performance. Can you think of a way to write this that doesn't recompile the regex every iteration?